test/conftest.py

```diff
@@ -0,0 +1,89 @@
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""System/E2E test setup"""
+import logging
+import subprocess
+from pathlib import Path
+
+import pytest
+
+pytest_plugins = "pytest_pixl"
+
+
+@pytest.fixture()
+def host_export_root_dir():
+    """Intermediate export dir as seen from the host"""
+    return Path(__file__).parents[1] / "exports"
+
+
+def run_subprocess(
+    cmd: list[str], working_dir: Path, *, shell=False, timeout=360
+) -> subprocess.CompletedProcess[bytes]:
+    """
+    Run a command but capture the stderr and stdout better than the CalledProcessError
+    string representation does
+    """
+    try:
+        logging.info("Running command %s", cmd)
+        return subprocess.run(
+            cmd,
+            check=True,
+            cwd=working_dir,
+            shell=shell,  # noqa: S603 input is trusted
+            timeout=timeout,
+            capture_output=True,
+        )
+    except subprocess.CalledProcessError as exception:
+        logging.error("*** exception occurred running: '%s'", cmd)  # noqa: TRY400 will raise anyway
+        logging.error("*** stdout:\n%s", exception.stdout.decode())  # noqa: TRY400
+        logging.error("*** stderr:\n%s", exception.stderr.decode())  # noqa: TRY400
+        raise
+
+
+TEST_DIR = Path(__file__).parent
+RESOURCES_DIR = TEST_DIR / "resources"
+RESOURCES_OMOP_DIR = RESOURCES_DIR / "omop"
+
+
+@pytest.fixture(scope="session")
+def _setup_pixl_cli(ftps_server) -> None:
+    """Run pixl populate/start. Cleanup intermediate export dir on exit."""
+    # CLI calls need to have CWD = test dir so they can find the pixl_config.yml file
+    run_subprocess(["pixl", "populate", str(RESOURCES_OMOP_DIR.absolute())], TEST_DIR)
+    run_subprocess(["pixl", "start"], TEST_DIR)
+    # poll here for two minutes to check for imaging to be processed, printing progress
+    yield
+    run_subprocess(
+        [
+            "docker",
+            "exec",
+            "system-test-ehr-api-1",
+            "rm",
+            "-r",
+            "/run/exports/test-extract-uclh-omop-cdm/",
+        ],
+        TEST_DIR,
+    )
+
+
+@pytest.fixture(scope="session")
+def _extract_radiology_reports(_setup_pixl_cli) -> None:
+    """
+    run pixl extract-radiology-reports.
+    TODO: should then poll for two minutes to ensure that the database is populated,
```

Comment:
Could use a similar strategy as in https://github.com/UCLH-Foundry/PIXL/blob/545792b497975e4463f8afe970bae071b5f63884/test/scripts/check_max_storage_in_orthanc_raw.py#L43-L52? Maybe turn this in another helper in the `pytest` plugin?

Reply:
Yes, this pattern crops up a few times, although unfortunately with slight variation so de-duplicating it will involve a bit of testing.

Reply:
I have made a generic waiting for condition method now.